### PR TITLE
token 2022: add `UpdateGroupMaxSize` instruction from SPL Token Group interface

### PR DESCRIPTION
**Pull request by buffalojoec**
This PR adds the `UpdateGroupMaxSize` instruction to Token2022 - from the SPL Token Group interface.

### CODE REVIEW

**Comment by buffalojoec** [comment]
Current dependencies on/for this PR:
* master
  * **PR #5616** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5616" target="_blank">Graphite</a> 
    * **PR #5548** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5548" target="_blank">Graphite</a> 
      * **PR #5621** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5621" target="_blank">Graphite</a> 
        * **PR #5601** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5601" target="_blank">Graphite</a> 
          * **PR #5626** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5626" target="_blank">Graphite</a>  👈
      * **PR #5614** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5614" target="_blank">Graphite</a> 
      * **PR #5549** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5549" target="_blank">Graphite</a> 
        * **PR #5593** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5593" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5626?utm_source=stack-comment).

**Comment by joncinque** [comment]
Once these PRs are rebased, I'll take a look!

**Review by joncinque — DISMISSED** [review]
Just the one comment about the test, then this is good to go!

large-small

File: token/program-2022-test/tests/token_group_update_max_size.rs [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+            // 167..170:    extension discriminator (GroupPointer)
+            // 171..202:    authority
+            // 203..234:    group pointer
+            // 235..238:    extension discriminator (TokenGroup)
+            // 239..270:    mint
+            // 271..302:    update_authority
+            // 303..306:    size
+            // 307..310:    max_size
+            let (front, back) = old_data.split_at(302);
+            let (_, back) = back.split_at(3);
```
[the remark this answers is not in the document]
Comment (buffalojoec) [reply token/program-2022-test/tests/token_group_update_max_size.rs:132]: Ah nuts good catch. But why does it still pass? 😅

[the remark this answers is not in the document]
Comment (joncinque) [reply token/program-2022-test/tests/token_group_update_max_size.rs]: I spent a little bit of time trying to figure it out, but decided it was better to just point it out haha

**Pull request merged**